lib/core.js

```diff
@@ -48,51 +48,36 @@ module.exports = {
     classData.componentWillReceiveProps = function(nextProps) {
       var chart = this.state.chart;
 
-      // // Reset the array of datasets
-      chart.data.datasets.forEach(function(set, setIndex) {
-        set.data.forEach(function(val, pointIndex) {
-          set.data = [];
-        });
-      });
-
-      // // Reset the array of labels
-      chart.data.labels = [];
+      if (nextProps.redraw) {
+        chart.destroy();	// Reset the array of datasets
+        this.initializeChart(nextProps);
+      } else {
+        // assign all of the properites from the next datasets to the current chart
+        nextProps.data.datasets.forEach(function(set, setIndex) {
+          var chartDataset = chart.data.datasets[setIndex];
 
-      // Adds the datapoints from nextProps
-      nextProps.data.datasets.forEach(function(set, setIndex) {
-        set.data.forEach(function(val, pointIndex) {
-          chart.data.datasets[setIndex].data[pointIndex] = nextProps.data.datasets[setIndex].data[pointIndex];
+          for (var property in set) {
+            if (set.hasOwnProperty(property)) {
+              chartDataset[property] = set[property];
+            }
+          }
         });
-      });
 
-      // Sets the labels from nextProps
-      nextProps.data.labels.forEach(function(val, labelIndex) {
-          chart.data.labels[labelIndex] = nextProps.data.labels[labelIndex];
-      });
+        chart.data.labels = nextProps.data.labels;
 
-      // Updates Chart with new data
-      chart.update();
+        chart.update();
+      }
   };
 
     classData.initializeChart = function(nextProps) {
       var Chart = require('chart.js');
       var el = ReactDOM.findDOMNode(this);
       var ctx = el.getContext("2d");
 
-      if (chartType === 'PolarArea'){
-        var chart = new Chart(ctx, {
-          type: 'polarArea',
-          data: nextProps.data,
-          options: nextProps.options
-        });
-      } else {
-        var chart = new Chart(ctx, {
-          type: chartType.toLowerCase(),
-          data: nextProps.data,
-          options: nextProps.options
-        });
-      }
-      this.state.chart = chart;
+      this.state.chart = Chart[chartType](ctx, {
+        data: nextProps.data,
+        options: nextProps.options
+      });
```

Comment:
It also might not be possible to use setState since it is asynchronous, and the chart being available is required to be sync the way the code is currently written. I'd rather not turn this PR into a full re-factor.